app/src/androidTest/java/ch/epfl/sweng/SDP/auth/LoginActivityTest.java

```diff
@@ -37,51 +49,100 @@
     private final Instrumentation.ActivityMonitor monitor = getInstrumentation()
             .addMonitor(AccountCreationActivity.class.getName(), null, false);
 
+    // Add a monitor for the HomeActivity
+    private final Instrumentation.ActivityMonitor homeMonitor = getInstrumentation()
+            .addMonitor(HomeActivity.class.getName(), null, false);
+
     private IdpResponse mockIdpResponse;
-    private LoginActivity loginActivity;
+    private Intent mockIntent;
+    private Activity loginActivity;
 
     /**
      * Initializes the mock objects.
      */
     @Before
     public void init() {
-        mockIdpResponse = Mockito.mock(IdpResponse.class);
         loginActivity = activityRule.getActivity();
+        mockIdpResponse = Mockito.mock(IdpResponse.class);
+        mockIntent = Mockito.mock(Intent.class);
+        Mockito.when(mockIdpResponse.getEmail()).thenReturn("testEmail");
+        Mockito.when(mockIntent.getParcelableExtra(ExtraConstants.IDP_RESPONSE))
+                .thenReturn(mockIdpResponse);
     }
 
     @Test
     public void testFailedLoginNullResponse() {
-        Intent mockIntent = Mockito.mock(Intent.class);
         Mockito.when(mockIntent.getParcelableExtra(ExtraConstants.IDP_RESPONSE))
                 .thenReturn(null);
-        loginActivity.onActivityResult(42, 0, mockIntent);
+        activityRule.getActivity().onActivityResult(42, 0, mockIntent);
```

Comment:
Why aren't you using loginActivity anymore, because the activity can change ?

Reply:
If I do that I get an error "onActivityResult has protected access". No idea why that happens.

Reply:
Have you tried with `private LoginActivity loginActivity` when defining it, as it was before? If it doesn't work the variable is useless and we can remove it

Reply:
That was actually the problem, thanks.

Reply:
Here you forgot to use loginActivity haha